tritonbench/components/power/chart.py

```diff
@@ -0,0 +1,134 @@
+
+        for row in reader:
+            for col, value in zip(header, row):
+                if value == "[N/A]":
+                    logger.warning(
+                        f"[tritonbench][power] {col} is not available, skipping"
+                    )
+                    value = 0.0
+                else:
+                    value = (
+                        float(value)
+                        if col
+                        not in [
+                            "clocks_event_reasons.hw_thermal_slowdown",
+                            "clocks_event_reasons.sw_thermal_slowdown",
+                        ]
+                        else value
+                    )
+                data[col].append(value)
+
+    # Generate synthetic time axis (100 ms per sample)
+    n_samples = len(next(iter(data.values())))
+    time = [i * 0.1 for i in range(n_samples)]  # seconds (0.1s = 100 ms)
+
+    # Plot power chart
+    plt.figure(figsize=(10, 6))
+    for power_col in header[:3]:
+        plt.plot(time, data[power_col], label=power_col)
+    plt.xlabel("Time (s)")
+    plt.ylabel("Power (W)")
+    plt.legend()
+    plt.title(
+        f"[tritonbench] {benchmark_name} power consumption over time on {device_name}"
+    )
+    plt.savefig(
+        os.path.join(POWER_OUTPUT_DIR, "power.png"), dpi=300, bbox_inches="tight"
+    )
+    # Plot temp chart
+    plt.figure(figsize=(10, 6))
+    for temp_col in header[3:5]:
+        plt.plot(time, data[temp_col], label=temp_col)
+        plt.xlabel("Time (s)")
+        plt.ylabel("Temperature (C)")
+    plt.legend()
+    plt.title(f"[tritonbench] {benchmark_name} temperature over time on {device_name}")
+    plt.savefig(
+        os.path.join(POWER_OUTPUT_DIR, "temp.png"), dpi=300, bbox_inches="tight"
+    )
+    # Plot frequency chart
+    plt.figure(figsize=(10, 6))
+    for temp_col in header[5:7]:
+        plt.plot(time, data[temp_col], label=temp_col)
+        plt.xlabel("Time (s)")
+        plt.ylabel("Frequency (MHz)")
+    plt.legend()
+    plt.title(f"[tritonbench] {benchmark_name} frequency over time on {device_name}")
+    plt.savefig(
+        os.path.join(POWER_OUTPUT_DIR, "freq.png"), dpi=300, bbox_inches="tight"
+    )
+
+
+def power_chart_begin(benchmark_name, output_dir):
+    # check no other proc is running
+    global QUERY_PROC, POWER_OUTPUT_DIR
+    assert QUERY_PROC is None, "Power query process must be None to start a new one"
+    # clean up the directory
+    POWER_OUTPUT_DIR = os.path.join(output_dir, benchmark_name)
+    if not os.path.exists(POWER_OUTPUT_DIR):
+        os.mkdir(POWER_OUTPUT_DIR)
+    stdout_file_path = os.path.join(POWER_OUTPUT_DIR, QUERY_STDOUT_FILE)
+    stderr_file_path = os.path.join(POWER_OUTPUT_DIR, QUERY_STDERR_FILE)
+    # Run the command
+    query_cmd = QUERY_COMMAND.format(
+        QUERY_FREQUENCY=QUERY_FREQUENCY, QUERY_DEVICE=_get_cuda_device_id()
+    ).split(" ")
+    with open(stdout_file_path, "w") as stdout_file, open(
```

Comment:
nit for future improvement:
it's better to read the output directly from stdout/stderror. the log file dumps can be async and for backup.